test/new-e2e/tests/otel/utils/pipelines_utils.go

```diff
@@ -617,6 +581,66 @@ func createCalendarApp(ctx context.Context, s OTelTestSuite) {
 	require.NoError(s.T(), err, "Could not properly start deployment")
 }
 
+func getCalendarAppEnvVars(name string, otlpEndpoint string, ust bool) []corev1.EnvVar {
+	envVars := []corev1.EnvVar{{
+		Name:  "OTEL_CONTAINER_NAME",
+		Value: name,
+	}, {
+		Name:      "OTEL_K8S_NAMESPACE",
+		ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"}},
+	}, {
+		Name:      "OTEL_K8S_NODE_NAME",
+		ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "spec.nodeName"}},
+	}, {
+		Name:      "OTEL_K8S_POD_NAME",
+		ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.name"}},
+	}, {
+		Name:      "OTEL_K8S_POD_ID",
+		ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.uid"}},
+	}, {
+		Name:  "OTEL_EXPORTER_OTLP_ENDPOINT",
+		Value: otlpEndpoint,
+	}, {
+		Name:  "OTEL_EXPORTER_OTLP_PROTOCOL",
+		Value: "grpc",
+	}}
+	resourceAttrs := "k8s.namespace.name=$(OTEL_K8S_NAMESPACE)," +
+		"k8s.node.name=$(OTEL_K8S_NODE_NAME)," +
+		"k8s.pod.name=$(OTEL_K8S_POD_NAME)," +
+		"k8s.pod.uid=$(OTEL_K8S_POD_ID)," +
+		"k8s.container.name=$(OTEL_CONTAINER_NAME)," +
+		"host.name=$(OTEL_K8S_NODE_NAME)," +
+		fmt.Sprintf("%v=%v", customAttribute, customAttributeValue)
+
+	// Use Unified Service Tagging env vars instead of OTel env vars
+	if ust {
+		return append(envVars, []corev1.EnvVar{{
+			Name:  "DD_SERVICE",
+			Value: calendarService,
+		}, {
+			Name:  "DD_ENV",
+			Value: env,
+		}, {
+			Name:  "DD_VERSION",
+			Value: version,
+		}, {
+			Name:  "OTEL_RESOURCE_ATTRIBUTES",
+			Value: resourceAttrs,
+		}}...)
+	}
+
+	return append(envVars, []corev1.EnvVar{{
+		Name:  "OTEL_SERVICE_NAME",
+		Value: calendarService,
+	}, {
+		Name: "OTEL_RESOURCE_ATTRIBUTES",
+		Value: resourceAttrs +
+			",service.name=$(OTEL_SERVICE_NAME)," +
```

Comment:
nit: u don't need to set in resource attrs once you have set `OTEL_SERVICE_NAME`